src/govuk/components/exit-this-page/exit-this-page.test.js

```diff
@@ -0,0 +1,169 @@
+/**
+ * @jest-environment puppeteer
+ */
+
+const { goToComponent, goToExample } = require('govuk-frontend-helpers/puppeteer')
+
+const buttonClass = '.govuk-js-exit-this-page-button'
+const skiplinkClass = '.govuk-js-exit-this-page-skiplink'
+const overlayClass = '.govuk-exit-this-page-overlay'
+
+describe('/components/exit-this-page', () => {
+  describe('when JavaScript is unavailable or fails', () => {
+    beforeAll(async () => {
+      await page.setJavaScriptEnabled(false)
+    })
+
+    afterAll(async () => {
+      await page.setJavaScriptEnabled(true)
+    })
+
+    it('navigates to the href of the button', async () => {
+      await goToComponent(page, 'exit-this-page')
+
+      const href = await page.evaluate((buttonClass) => document.querySelector(buttonClass).href, buttonClass)
```

Comment:
Suggestion: Puppeteer has a handy [`$eval` function](https://pptr.dev/api/puppeteer.page._eval) that could be used to simplify this (and similar statements throughout these tests) slightly:

```suggestion
      const href = await page.$eval(buttonClass, el => el.href)
```